### PR TITLE
Feat: Migrate Infra from Pagoda to Sig Network

### PR DESCRIPTION
**Pull request by auto-mausx**

### CODE REVIEW

File: chain-signatures/contract/src/lib.rs [hunk 1]
```diff
@@ -611,7 +611,7 @@ impl VersionedMpcContract {
 
     // This function can be used to transfer the MPC network to a new contract.
     #[private]
-    #[init]
+    #[init(ignore_state)]
```
Comment (ppca) [inline chain-signatures/contract/src/lib.rs]: let's remove this change

File: infra/modules/leader/main.tf [hunk 2]
```diff
@@ -11,7 +11,7 @@ resource "google_cloud_run_v2_service" "leader" {
 
     vpc_access {
       connector = var.connector_id
-      egress    = "PRIVATE_RANGES_ONLY"
+      egress    = "ALL_TRAFFIC"
```
Comment (ppca) [inline infra/modules/leader/main.tf:14]: do we still want this?

File: infra/modules/signer/main.tf [hunk 3]
```diff
@@ -13,7 +13,7 @@ resource "google_cloud_run_v2_service" "signer" {
       for_each = var.connector_id == null ? [] : [1]
       content {
         connector = var.connector_id == null ? null : var.connector_id
-        egress    = "PRIVATE_RANGES_ONLY"
+        egress    = "ALL_TRAFFIC"
```
Comment (ppca) [inline infra/modules/signer/main.tf:16]: same here

File: infra/mpc-recovery-prod/main.tf [hunk 4]
```diff
@@ -43,14 +39,14 @@ resource "google_service_account" "service_account" {
   display_name = "MPC Recovery mainnet Account"
 }
 
-resource "google_service_account_iam_binding" "serivce-account-iam" {
-  service_account_id = google_service_account.service_account.name
-  role               = "roles/iam.serviceAccountUser"
+# resource "google_service_account_iam_binding" "serivce-account-iam" {
+#   service_account_id = google_service_account.service_account.name
+#   role               = "roles/iam.serviceAccountUser"
```
Comment (ppca) [inline infra/mpc-recovery-prod/main.tf]: remove if not necessary?

File: infra/mpc-recovery-testnet/main.tf [hunk 5]
```diff
@@ -43,14 +39,14 @@ resource "google_service_account" "service_account" {
...
-  role               = "roles/iam.serviceAccountUser"
+# resource "google_service_account_iam_member" "serivce-account-iam" {
+#   service_account_id = google_service_account.service_account.name
+#   role               = "roles/iam.serviceAccountUser"
 
-  members = [
-    "serviceAccount:${local.client_email}",
-  ]
-}
+#   member = [
```
Comment (ppca) [inline infra/mpc-recovery-testnet/main.tf]: delete if not needed

File: infra/multichain-dev/main.tf [hunk 6]
```diff
@@ -83,29 +91,50 @@ module "gce-container" {
...
   service_account = {
-    email  = "mpc-recovery@pagoda-discovery-platform-dev.iam.gserviceaccount.com",
+    email  = google_service_account.service_account.email,
     scopes = ["cloud-platform"]
   }
   name_prefix          = "multichain-${count.index}"
   source_image_family  = "cos-113-lts"
   source_image_project = "cos-cloud"
-  machine_type         = "n2-standard-2"
+  machine_type         = "e2-medium"
```
Comment (ppca) [inline infra/multichain-dev/main.tf:137]: is this cheaper now compared with n2-standard-2?

**Review by ppca — DISMISSED** [review]
lgtm except a couple comments

File: infra/multichain-dev/main.tf (hunk 6 above)
Comment (auto-mausx) [reply infra/multichain-dev/main.tf:137]: Considerably yes, we could probably do even cheaper.

File: infra/modules/leader/main.tf (hunk 2 above)
Comment (auto-mausx) [reply infra/modules/leader/main.tf:14]: yes, since we want all traffic to leave our VPC.

File: infra/modules/signer/main.tf (hunk 3 above)
Comment (auto-mausx) [reply infra/modules/signer/main.tf:16]: Also yes since technically signers are available publicly for our leader node.

**Review by ppca — APPROVED** [review]

File: infra/multichain-dev/main.tf (hunk 6 above)
Comment (volovyks) [reply infra/multichain-dev/main.tf:137]: I do not think this is the reason, but our dev network is slower now. Let's keep the hardware in mind.

**Pull request merged**